### ISSUE TITLE
[Bug]: Currency Input widget issues

### ISSUE THREAD

**Issue by shadabbuchh** [appsmithorg/appsmith#14835]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When user changes the `Decimals` back to `0`, the value does not automatically round off rather the user has to click inside the widget for the value to round off.
https://www.loom.com/share/3257dda3e9b4459a9bf8f49a9bfa280c

### Steps To Reproduce

1. D&D a `Currency Input` widget.
2. Set the `Decimals` to `2`.
3. Enter any value with decimals.
4. Now change the `Decimals` value back to `0`.

Notice that the value does not automatically round off to the nearest number. The user has to click inside the widget in order to round off the value.

### Version

Cloud

**Comment by dilippitchika** [comment]
Hey @codedmind this is a low priority item on our plate and I don't think we will get to it anytime soon. Why is this important for you?

**Comment by codedmind** [comment]
@dilippitchika one of our apps have a lot of currency widgets and we are not able to input decimals values... so this have a high impact in our end.
Are this difficult to solve? Otherwise we must redo the app and replace the currency widget...

**Comment by dilippitchika** [comment]
@codedmind this is a different issue, i believe you are talking about this issue - https://github.com/appsmithorg/appsmith/issues/14261 - where the decimal separator is not working in EU countries as expected.

This is a different issue, which only happens in the edit mode and has no impact on view mode

**Comment by codedmind** [comment]
@dilippitchika your are correct! My mistake ... any news regarding that issue :D

**Comment by dilippitchika** [comment]
We have figured out a solution, but are stuck in terms of scaling that. Could you help us with the details of how your users use currencies, do all of them follow a single style for entering decimals like . or based on the country they are located in they change the way they write decimals?

**Comment by codedmind** [comment]
@dilippitchika yes they all use the same way... in our case the decimal is the comma

**Comment by PrasadMadine** [comment]
Hi @shadabbuchh , I'm picking this issue.

**Comment by PrasadMadine** [comment]
Hi @shadabbuchh , the selected decimals are not being reflected even on onFocus, it's reflecting while clicking on onFocus and then clicking on the canvas.
